### ISSUE TITLE
Consider migrating translations from weblate.bubu1.eu to another Weblate instance

### ISSUE THREAD

**Issue by Linerly** [searxng/searxng#2076]
![Screenshot of weblate.bubu1.eu showing the shutdown announcement and doesn't accept new projects anymore](https://user-images.githubusercontent.com/47475676/210829782-a95a6a54-4857-44e4-90d4-831aff6fdd9d.png)

**Comment by return42** [comment]
Thanks for pointing out. Its not yet spoken out but we consider to switch to https://translate.codeberg.org

**Comment by comradekingu** [comment]
@return42 Please avoid Codeberg. It now has a CoC, meaning the only reason to use it over any alternative is gone.
For translations, https://hosted.weblate.org/hosting/ has better terms, and a more active userbase.

**Comment by return42** [comment]
> Please avoid Codeberg. It now has a CoC, meaning the only reason to use it over any alternative is gone.

Sorry, I don't see the issue with codeberg / can you explain?

> For translations, https://hosted.weblate.org/hosting/ has better terms,

can you give some hints about the pro & cons you think of?

> and a more active userbase.

That argument counts on the *pros* side for weblate.org.

In an internal discussion I voted for Codeberg since it is a registered society (not a company that can be sold) and the server are located in DE (Europe & GDPR).

In my weighing of the pros and cons, codeberg wins ... have I perhaps the wrong criteria or overlooked something important?

**Comment by Linerly** [comment]
> Thanks for pointing out. Its not yet spoken out but we consider to switch to https://translate.codeberg.org

Now that it's migrated to Codeberg Translate, consider editing [the comment](https://github.com/searxng/searxng/issues/540#issue-1061627572) in #540 with the updated link (https://translate.codeberg.org/projects/searxng/searxng/)

Also, the project seems to be locked:
![A screenshot of the Weblate page for SearXNG, with the message that reads “The translation is temporarily closed for contributions due to maintenance, please come back later.”](https://user-images.githubusercontent.com/47475676/213923368-c7afcfbe-e5ec-4179-ab52-861cd5910135.png)

**Comment by comradekingu** [comment]
@return42
Codeberg doesn't facilitate sharing of strings between projects in any way.
It then relies purely on licenses, and that doesn't work (as Canonical found out with Launchpad).
This and more pitfalls are not avoided by having a legal document similar to
https://hosted.weblate.org/legal/terms/.
EU-GDPR sounds a lot better when actually making some effort to comply.
Where do I find out who the DPO is for example?
Instead it has just a very general provision:
https://codeberg.org/Codeberg/org/src/branch/main/Imprint.md

Moreover, Codeberg has in the past allowed non-free licenses, which are still present on its server,
so it isn't protecting any ground from an ethical standpoint.
This is IMO not a good precedence for its ability to make sound decisions.

Effectively, like-for-like, an active userbase of more and better translators and similar projects is the value you can hope to expect from a hosting platform.
Relying on it in the form of Weblate is no less true as it in and of itself has created the tool with which to organise translation.
From a profit-motive, Weblate has created this, and sees it in its interest to help libre software.

Codeberg on the other hand is ruled by a committee, that in addition to operating on a product redundant to the powers of Weblate, answer to no particular directives other than keeping its tax-exempt status.
If one trusts in its reputation and goodwill that come from from people picking the lesser option to grant more freedom,
(like my recommendation here to use BuBu) we find Codeberg used it to land a CoC.
The country it is registered in is also happily eroding free speech.

There is avoiding going with the biggest players and granting smaller efforts some help, and then
there is misplaced trust.
Do you trust yourselves not to take the contributions of others (for example mine), and have them appear as if they are made in support of a CoC? You are as free to do this as BuBu and Codeberg, or anyone else is.

The total risk in a first-party platform can be summed up in Weblate facilitating migration, just as much as it doesn't hold any serious assets that aren't GPLv3-licensed
(without it being subject to DCOs, CLAs, etc).

Avoiding potential downfall where possible is a laudable aim, but one that would suggest
a proprietary platform owned by Microsoft isn't how to best do what can be done just as well elsewhere.
That place still isn't Codeberg.

**Comment by return42** [comment]
> Now that it's migrated to Codeberg Translate, consider editing https://github.com/searxng/searxng/issues/540#issue-1061627572 in https://github.com/searxng/searxng/issues/540 with the updated link (https://translate.codeberg.org/projects/searxng/searxng/)

Thanks :+1: / fixed.

> Also, the project seems to be locked:

@dalf are there any reason why you have locked (3 hours ago)?

**Comment by comradekingu** [comment]
Might it be to ask contributors what sort of ethical considerations their contributions now represent?
Considering they are taken without compensation that only seems fair.

You haven't migrated it with the files, so all traces of who did what is gone.
Practically I additionally care more about quality translations than that, but alas.

**Comment by return42** [comment]
@comradekingu please give us some time to discuss your arguments internally .. 

We were in need of action and just changed quickly, with Codeberg we had problems to import a backup, so we just took the files.

We are grateful when people point out mistakes, we always try to improve, but with the few people we are we can't always do everything perfectly ... we make mistakes and sometimes it would be better to take more time to weigh a decision fairly / always knowing that we can never meet all the demands.  But sometimes we have to act quickly and then we choose the easiest option for us.  But that doesn't mean that we can't change and improve things afterwards.

**Comment by comradekingu** [comment]
@return42 
I have deep respect for your reconsideration, and I am offering my time in setting it up and fixing things if you add "kingu" as a Weblate admin on Hosted. A self-hosted instance listed on https://weblate.org/discover/ is an option to consider.
I did give you a hard time here, but I equally do what is in my power to help laudable efforts.
Starting from scratch isn't a deal-breaker. It largely comes down to that to arrive at a better stringbase anyway.

**Comment by dalf** [comment]
@comradekingu the project has been created on hosted.weblate.org. I've sent the request for Libre hosting one week ago, but the request is still pending.
I'm not sure what to do more.

**Comment by return42** [comment]
@comradekingu https://hosted.weblate.org/projects/searxng/searxng/

**Comment by comradekingu** [comment]
@return42 
Top stuff. If you add "kingu" as an admin in https://hosted.weblate.org/access/searxng/#users I can try to fix the error that causes all the source strings to be blank.

**Comment by return42** [comment]
@comradekingu did you read the post from @dalf

> the project has been created on hosted.weblate.org. I've sent the request for Libre hosting one week ago, but the request is still pending.
> I'm not sure what to do more.

More and more I have a doubt it is a good decision to move to hosted.weblate.org.

> I can try to fix the error that causes all the source strings to be blank.

Thanks, but such issues .. and that we are so dependent on the mercy of the hoster is quite strange to me and I have my doubts.

At codeberg we could have started weeks ago without any mercy, issues and other noise.

**Comment by comradekingu** [comment]
@return42
A requirement for the libre hosting is that the setup is correct.
I think the error is a discrepancy between the two in
"Template for new translations" at the bottom of
https://hosted.weblate.org/settings/searxng/searxng/#files
I don't think it is a bug in Weblate 4.16 vs 4.15.1

There are less than two orders of magnitude fewer users on Codeberg Weblate.
Those users have yet to make it to 50 translated strings the last month, when also counting changes.
https://translate.codeberg.org/changes/?page=3&limit=20&project=searxng&action=2&action=5&start_date=2022-12-30&end_date=2023-02-10 (and many of those are identical to the source strings).
Weblate as a host has a longer and better history, knows more about Weblate, with more services on offer, and always runs the latest version.
How good the Codeberg economy is I don't know, but continued donations to a lesser code hosting service that lands a CoC outside of community control and doesn't have a firm grip on licensing seems like an equal dependency, on more shaky ground. Not to mention legally there is no legal framework for sharing of strings, etc.

I answered within an hour and offered to help.
There are plenty other things I can do to help with that access, which also works in your favour.

**Comment by dalf** [comment]
> A requirement for the libre hosting is that the setup is correct.

Does it require providing push access to the master branch?
Currently, the translations are pull from the Weblate.

**Comment by comradekingu** [comment]
No, you can set it up to just generate PRs. :)
It isn't a requirement, but
https://docs.weblate.org/no/latest/admin/continuous.html#github-setup is on the checklist.
You could instead just add the Weblate user https://github.com/WeblateOrg/ to the organization (without the write access).
I think it will work with just the push-address too.

The method of integration can be selected at the top of
https://hosted.weblate.org/settings/searxng/searxng/#vcs
where it says "version control system"
"GitHub with merge requests" would be the one to pick.

It is possible to select how often to make those in
"Age of changes to commit" at the bottom of the same page.
And there is a "Squash Git commits" plugin to reduce the noise
in https://hosted.weblate.org/addons/searxng/searxng/

If you want none of that and want to do it all manually from https://hosted.weblate.org/projects/searxng/searxng/#repository  each time, I think that should be OK(?). Mention it in the request when sending in the request for approval.

The idea is to not have translations stay on Weblate only, and ultimately prevent effort going into translations from not getting used.

**Comment by return42** [comment]
> @comradekingu the project has been created on hosted.weblate.org. I've sent the request for Libre hosting one week ago, but the request is still pending.

https://hosted.weblate.org/projects/searxng/

![grafik](https://user-images.githubusercontent.com/554536/219580344-296e700f-9279-41df-af39-26a73953cd62.png)

@comradekingu do you see a chance to force the process?

I follow your arguments to move to hosted.weblate.org and I want to move .. BUT .. our translation workflow becomes more and more a mess while we are waiting to get accepted as a libre project .. if we won't accepted in the next days we have to switch back to codeberg in order not to sink in an absolute chaos ... I also do not know any more how I should explain this to the users still.

**Comment by comradekingu** [comment]
@return42 
It doesn't have to be accepted to be used. The message only says for translators to take precautions since some projects do get removed.
The fraudulent ones from last round are still up, and a few filed before SearXNG have it still.
It just means the person responsible hasn't gotten around to it yet this week.

I couldn't get access to the Weblate administration, so I hope that part is OK.
As long as the request for approval is in, everything should be good to go.
The #2200 PR can be merged, and the Codeberg effort locked.
Posting an announcement there before taking it down is also a good idea.

**Comment by return42** [comment]
> The https://github.com/searxng/searxng/pull/2200 PR can be merged, and the Codeberg effort locked.

There is more work to do / see a9d6f7532a

--- 

> It doesn't have to be accepted to be used. The message only says for translators to take precautions since some projects do get removed.
> As long as the request for approval is in, everything should be good to go.

Until SearXNG is approved, we will not direct translators to this platform.

**Comment by dalf** [comment]
I'm sorry, I'm going to postpone the usage of hosted.weblate.org for now.

Currently, there are some shell scripts to pull the new translations: the Weblate instance does not have access to the git repository. It was implemented this way because we didn't knew weblate.bubu1.eu . I felt uncomfortable giving access to the git repository, especially since a lot of instances pull right away the newly built version from the master branch.

Currently, I lack of energy to update the process for hosted.weblate.org. I've tried and mess up, I may try later.
For now the translations are going to stay on codeberg because it works despite what was said above.

Thank you @comradekingu for your help and your time and sorry.
